bazelrunner/src/main/java/org/jetbrains/bsp/bazel/bazelrunner/BazelInfo.kt

```diff
@@ -1,16 +1,31 @@
 package org.jetbrains.bsp.bazel.bazelrunner
 
+import java.lang.RuntimeException
 import java.nio.file.Path
 
 interface BazelInfo {
   val execRoot: String
   val workspaceRoot: Path
+  val release: BazelRelease
 }
 
-
+data class BazelRelease(
+        val versionString: String
+) {
+  val major: Int by lazy {
+    """(?<=release )\d+(?=[0-9.]*)""".toRegex().find(versionString)?.value?.toInt()!!
+  }
+
+  fun mainRepositoryReferencePrefix() = when(major) {
+    in 0..3 -> throw RuntimeException("Unsupported Bazel version, use Bazel 4 or newer")
+    in 4..5 -> "//"
+    else -> "@//"
+  }
```

Comment:
can we have tests for it? including this issue as well: https://github.com/JetBrains/bazel-bsp/issues/92